src/core/Form/Textarea/Textarea.tsx

```diff
@@ -1,52 +1,162 @@
+import React, { Component, ChangeEvent, FocusEvent } from 'react';
 import { default as styled } from 'styled-components';
+import classnames from 'classnames';
 import { TokensProp, InternalTokensProp } from '../../theme';
-import {
-  Textarea as CompTextarea,
-  TextareaProps as CompTextareaProps,
-} from '../../../components/Form/Textarea';
 import { baseStyles } from './Textarea.baseStyles';
 import { withSuomifiDefaultProps } from '../../theme/utils';
-import classnames from 'classnames';
+import { HtmlTextarea, HtmlTextareaProps, HtmlDiv } from '../../../reset';
+import { AutoId } from '../../../utils/AutoId';
+import { LabelText } from '../LabelText/LabelText';
+import { HintText } from '../HintText/HintText';
+import { StatusText } from '../StatusText/StatusText';
+import { InputStatus } from '../types';
 
 const baseClassName = 'fi-textarea';
-
 const textareaClassNames = {
+  textareaContainer: `${baseClassName}_textarea-element-container`,
+  textarea: `${baseClassName}_textarea`,
   disabled: `${baseClassName}--disabled`,
   error: `${baseClassName}--error`,
 };
 
-export interface TextareaProps extends CompTextareaProps, TokensProp {}
+type TextareaStatus = Exclude<InputStatus, 'success'>;
 
-const StyledTextarea = styled(
-  ({ tokens, ...passProps }: TextareaProps & InternalTokensProp) => (
-    <CompTextarea {...passProps} />
-  ),
-)`
-  ${(props) => baseStyles(props)}
-`;
+export interface TextareaProps extends HtmlTextareaProps, TokensProp {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Disable usage */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: () => void;
+  /** To execute on textarea text change */
+  onChange?: (event: ChangeEvent<HTMLTextAreaElement>) => void;
+  /** To execute on textarea text onBlur */
+  onBlur?: (event: FocusEvent<HTMLTextAreaElement>) => void;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: 'hidden' | 'visible';
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Text content for textarea */
+  children?: string;
+  /** Hint text to be shown below the component */
+  hintText?: string;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: TextareaStatus;
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** Resize mode of the textarea
+      'both' | 'vertical' | 'horizontal' | 'none'
+      @default 'vertical' 
+   */
+  resize?: 'both' | 'vertical' | 'horizontal' | 'none';
+  /** Optional text that is shown after labelText. Will be wrapped in parentheses. */
+  optionalText?: string;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated automatically
+   */
+  id?: string;
+  /** Input name */
+  name?: string;
+}
```

Comment:
Do we need `fullWidth` prop for Textarea like with TextInput?

https://github.com/vrk-kpa/suomifi-ui-components/blob/de3ba041127f51eb2df5c75ff152933eefc16e13/src/core/Form/TextInput/TextInput.tsx#L71-L72

Reply:
Wouldn't hurt I guess. Not sure if there is a design need for this, but would improve usability at least and cover more use cases (e.g. switching to full available width in mobile view).

Reply:
I think it would be a welcome addition.

Reply:
`fullWidth` prop is now added to Textarea.